app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -1,20 +1,98 @@
 package com.jetbrains.handson.mpp.mobile
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.AlertDialog
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
-import android.widget.TextView
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.Button
+import android.widget.Spinner
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.departSelector
+
 
 class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
+    private var departStation: Station? = null
+    private var arrivalStation: Station? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         val presenter = ApplicationPresenter()
         presenter.onViewTaken(this)
+
+        val stationsHelper = StationsHelper()
+
+        val stations = stationsHelper.stations
+
+        populateSpinner(R.id.departSelector, stations)
+        populateSpinner(R.id.arrivalSelector, stations)
+
+        findViewById<Spinner>(R.id.departSelector).setOnItemSelectedListener(object : OnItemSelectedListener {
```

Comment:
Rather than findViewById have a look in to view binding, it's safer and importantly cleaner! Let me know if you want a hand with it :)